javaslang/src/main/java/javaslang/collection/PriorityQueue.java

```diff
@@ -0,0 +1,657 @@
+    }
+
+
+    @Override
+    public Tuple2<? extends PriorityQueue<T>, ? extends PriorityQueue<T>> span(Predicate<? super T> predicate) {
+        throw new UnsupportedOperationException();
+    }
+
+
+    @Override
+    public PriorityQueue<T> tail() {
+        return deleteMin();
+    }
+
+    @Override
+    public Option<? extends PriorityQueue<T>> tailOption() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public PriorityQueue<T> take(long n) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public PriorityQueue<T> takeRight(long n) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public PriorityQueue<T> takeUntil(Predicate<? super T> predicate) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public PriorityQueue<T> takeWhile(Predicate<? super T> predicate) {
+        throw new UnsupportedOperationException();
+    }
+
+
+    @Override
+    public <T1, T2> Tuple2<? extends PriorityQueue<T1>, ? extends PriorityQueue<T2>> unzip(Function<? super T, Tuple2<? extends T1, ? extends T2>> unzipper) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <T1, T2, T3> Tuple3<? extends Seq<T1>, ? extends Seq<T2>, ? extends Seq<T3>> unzip3(Function<? super T, Tuple3<? extends T1, ? extends T2, ? extends T3>> unzipper) {
+        throw new UnsupportedOperationException();
+    }
+
+
+    @Override
+    public <U> PriorityQueue<Tuple2<T, U>> zip(Iterable<? extends U> that) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public <U> PriorityQueue<Tuple2<T, U>> zipAll(Iterable<? extends U> that, T thisElem, U thatElem) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public PriorityQueue<Tuple2<T, Long>> zipWithIndex() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) {
+            return true;
+        } else if (o instanceof PriorityQueue) {
+            final PriorityQueue<?> that = (PriorityQueue<?>) o;
+            return this.toList().equals(that.toList());
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return toList().hashCode();
+    }
+
+    @Override
+    public String stringPrefix() {
+        return "PriorityQueue";
+    }
+
+    @Override
+    public String toString() {
+        return mkString(stringPrefix() + "(", ", ", ")");
+    }
+
+    public PriorityQueue<T> padTo(int length, T element) {
+        final int actualLength = length();
+        if (length <= actualLength) {
+            return this;
+        } else {
+            return ofAll(comparator, toList().padTo(length, element));
+        }
+    }
+}
+
+abstract class PriorityQueueBase<T, Q extends PriorityQueueBase<T, Q>> extends AbstractsQueue<T, Q> {
```

Comment:
`PriorityQueue` doesn't seem like a `Seq` as inserting to a given index doesn't really make sense.
Made it a `Traversable`